mirroring/mirroring.go

```diff
@@ -90,6 +91,42 @@ func GetLogInfo() (*models.LogInfo, error) {
 	return logInfoResp.GetPayload(), nil
 }
 
+func GetLogProof(firstSize *int64) (*models.ConsistencyProof, error) {
+	rekorServerURL := viper.GetString("rekorServerURL")
+	rekorClient, err := client.GetRekorClient(rekorServerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	logInfo, err := GetLogInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	params := tlog.NewGetLogProofParams()
+	if *firstSize > 1 {
+		params.FirstSize = firstSize
+	}
+	params.LastSize = *logInfo.TreeSize
+
+	logProofResp, err := rekorClient.Tlog.GetLogProof(params)
+	if err != nil {
+		return nil, err
+	}
+	return logProofResp.GetPayload(), nil
+}
+
+func VerifyConsistencyProof(rootHash *string, logProof *models.ConsistencyProof) bool {
```

Comment:
I'm not sure if this function is doing all it should, but we can discuss this. Shouldn't it be reconstructing the new roothash using the old hash and the proof